packages/core/http/core-http-server/src/router/raw_request.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { URL } from 'url';
+import type { Request, RouteOptionsPayload } from '@hapi/hapi';
+import type { KibanaRouteOptions } from './request';
+import type { Headers } from './headers';
+
+/**
+ * Represents a fake raw request.
+ * Can be used to instantiate a `KibanaRequest`.
+ */
+export interface FakeRawRequest {
+  /** The headers associated with the request. */
+  headers: Headers;
+  /** The path of the request */
+  path: string;
+  method?: string;
+  url?: URL;
+  app?: Record<string, unknown>;
+  auth?: {
+    isAuthenticated?: boolean;
+  };
+  route?: {
+    settings?: {
+      tags?: string[];
+      app?: KibanaRouteOptions;
+      payload?: RouteOptionsPayload;
+    };
+  };
+}
+
+export type RawRequest = Request | FakeRawRequest;
```

Comment:
The naming tripped me up a bit here. 

When importing `RawRequest` and `FakeRawRequest` it makes it seem like `RawRequest` is the real one and `FakeRawRequest` is the fake one when in fact `Request` is the real one and `RawRequest` is the combination of the of the two.

Given the existing types and `FakeRequest` already having a different meaning I'm not sure if there's a way to make the relationship clearer so happy to leave as is.

Reply:
Yeah... we have to many fake things related to requests, and the naming is obviously questionable. The fact that one of the type (`HAPI.Request`) is not directly ours doesn't help too. If anyone has a better naming proposal, I'll gladly take it.